rest_framework/compat.py

```diff
@@ -14,13 +13,6 @@ def unicode_http_header(value):
     return value
 
 
-def distinct(queryset, base):
```

Comment:
can we change this in a separate PR? or it is part of the change in this PR?

Reply:
It could be splitted, however this method was introduced in https://github.com/encode/django-rest-framework/commit/b4b2dc18fa4cb6bbe008d4f6e59b58862ed20be6 only for the search class and putted into `compact` package in the hope of being re-used.

IMO this should be kept togheter with the rest of the PR since this change goes in the same direction of "align the behaviour with django".

I just noticed that the code I used (which was introduced in https://github.com/django/django/commit/187118203197801c6cb72dc8b06b714b23b6dd3d) has just been removed from the main branch in https://github.com/django/django/commit/d569c1dcfeb26ca9ee391e5dfeadedf2b5ed4253. So the whole thing may just be changed to use `.distinct()` like django is going to do.

PS: I really do not understand why 8 years ago there was the need to have a different implementation of distinct only for Oracle engine and why this was implemented here.

Reply:
> PS: I really do not understand why 8 years ago there was the need to have a different implementation of distinct only for Oracle engine and why this was implemented here.

at that time django oracle back end was not properly maintained. but now it is